editor/header/tools/publish-button.js

```diff
@@ -18,42 +19,41 @@ import {
 	getPostEdits,
 	getBlocks,
 	isSavingPost,
-	didPostSaveRequestSucceed,
-	didPostSaveRequestFail,
-	isSavingNewPost,
+	isEditedPostAlreadyPublished,
+	isEditedPostBeingScheduled,
+	getEditedPostVisibility,
 } from '../../selectors';
 
 function PublishButton( {
 	post,
 	edits,
 	dirty,
 	blocks,
-	isSuccessful,
-	isRequesting,
-	isError,
-	requestIsNewPost,
+	isSaving,
+	isPublished,
 	onSave,
+	isBeingScheduled,
+	visibility,
 } ) {
-	const buttonEnabled = ! isRequesting;
-
+	const buttonEnabled = ! isSaving &&
```

Comment:
Thinking some of this more complex logic warrants testing, or moving to selector e.g. `isPostPublishable`.